runtime/resolvers/resolvers_test.go

```diff
@@ -0,0 +1,158 @@
+package resolvers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"maps"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/testruntime"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+type Project struct {
+	Sources    map[string]yaml.Node
+	Models     map[string]yaml.Node
+	Dashboards map[string]yaml.Node
+	APIs       map[string]yaml.Node
+}
+type Resolvers struct {
+	Project    Project
+	Connectors map[string]*testruntime.InstanceOptionsForResolvers
+	Tests      map[string]*Test
+}
+
+type Test struct {
+	Resolver string
+	Options  runtime.ResolveOptions
+	Result   []map[string]any
+}
+
+var update = flag.Bool("update", false, "Update test results")
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	os.Exit(m.Run())
+}
+
+func TestResolvers(t *testing.T) {
+
+	entries, err := os.ReadDir("./")
+	require.NoError(t, err)
+	var reg = regexp.MustCompile(`^(.*)_resolvers_test.yaml$`)
+	for _, e := range entries {
+		if reg.MatchString(e.Name()) {
+			t.Log("Running with", e.Name())
+			yamlFile, err := os.ReadFile(e.Name())
+			require.NoError(t, err)
+			var r Resolvers
+			err = yaml.Unmarshal(yamlFile, &r)
+			require.NoError(t, err)
+
+			files := make(map[string]string)
+			for name, node := range r.Project.Sources {
+				abs := filepath.Join("sources", name)
+				bytes, err := yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+			for name, node := range r.Project.Models {
+				abs := filepath.Join("models", name)
+				var bytes []byte
+				bytes, err = yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+			for name, node := range r.Project.Dashboards {
+				abs := filepath.Join("dashboards", name)
+				bytes, err := yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+			for name, node := range r.Project.APIs {
+				abs := filepath.Join("apis", name)
+				bytes, err := yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+
+			for connector, opts := range r.Connectors {
+				t.Log("Running with", connector)
+				if opts == nil {
+					opts = &testruntime.InstanceOptionsForResolvers{}
+				}
+				if opts.Files == nil {
+					opts.Files = map[string]string{"rill.yaml": ""}
+				}
+
+				switch connector {
+				case "druid":
+					opts.OLAPDriver = "druid"
+				case "clickhouse":
+					opts.OLAPDriver = "clickhouse"
+				}
+
+				maps.Copy(opts.Files, files)
+				rt, instanceID := testruntime.NewInstanceForResolvers(t, *opts)
+				for testName, test := range r.Tests {
+					t.Run(testName, func(t *testing.T) {
+						t.Log("======================")
+						t.Log("Running ", testName, "with", e.Name(), "and", connector)
+						testruntime.RequireParseErrors(t, rt, instanceID, nil)
```

Comment:
We should also check that there aren't any reconcile errors. Consider using `RequireReconcileState` and patching it to skip checks when the desired length is `-1`. Then you could call `RequireReconcileState(t, rt, instanceID, -1, 0, 0)`.

Reply:
Done.